### PR TITLE
[Feature Fix] Fix resolve icon view

### PR DESCRIPTION
**Pull request by leb2dg**
# Purpose

This fixes issues created by PR #2871. The certificate and collection icons on the project organizer are incorrect. Private component icons appear on the file grid.

# Changes

`resolveIconView` in *fangorn.js* is revamped to handle requests from the project organizer and the file grid. The change in `resolveIconView` eradicates the need for `_poResolveIcon` in *projectorganizer.js*. The project organizer now calls `resolveIconView` directly.

# Side effects

In the file grid, the private component icon is used. In the project organizer, the dashboard icon, 'all my registrations' and 'all my project icons' display properly.

Fix:
screen shot 2015-06-05 at 1 01 30 pm

screen shot 2015-06-05 at 1 01 06 pm

Problem:
screen shot 2015-06-05 at 9 58 30 am

screen shot 2015-06-05 at 1 03 59 pm

### CODE REVIEW

**Comment by leb2dg** [comment]
@nicipfeiffer @sloria This should be merged before testing for PR #2871 is tested.

**Comment by lbanner** [comment]
Nice @leb2dg :+1:

File: website/static/js/fangorn.js [hunk 1]
```diff
@@ -209,30 +213,30 @@ function resolveIconView(item) {
...
     }
-    if (item.data.nodeType === 'smartFolder') {
-        return returnView('smartCollection');
-    }
-    if (item.data.nodeType === 'folder') {
+    if (item.data.isDashboard) {
         return returnView('collection');
     }
-    if (item.data.nodeType === 'pointer' && item.parent().data.nodeType !== 'folder') {
+    if (item.data.nodeType === 'smartFolder' || item.data.isSmartFolder) {
```
Comment (sloria) [inline website/static/js/fangorn.js]: This seems incorrect; nodeType can neve be `smartFolder`; see https://github.com/CenterForOpenScience/osf.io/blob/33ebadfa50e1a92922113a32e0658bba05db7541/website/util/rubeus.py#L368

File: website/static/js/fangorn.js [hunk 2]
```diff
@@ -209,30 +213,30 @@ function resolveIconView(item) {
...
+        return returnView('smartCollection');
+    }
+    if ((item.data.nodeType === 'pointer' && item.parent().data.nodeType !== 'folder') || (item.data.isPointer && !item.parent().data.isFolder)) {
         return returnView('link');
     }
-    if (item.data.nodeType === 'project') {
+    if (item.data.nodeType === 'project' || item.data.isProject) {
```
Comment (sloria) [inline website/static/js/fangorn.js]: Why is the check for `isProject` necessary? Isn't that taken care of by `nodeType === 'project'`?

File: website/static/js/fangorn.js [hunk 3]
```diff
@@ -209,30 +213,30 @@ function resolveIconView(item) {
...
         if (item.data.isRegistration) {
             return returnView('registeredProject', item.data.category);
         } else {
             return returnView('project', item.data.category);
         }
     }
-    if (item.data.nodeType === 'component') {
+    if (item.data.nodeType === 'component' || item.data.isComponent) {
```
Comment (sloria) [inline website/static/js/fangorn.js]: Same here; is `isComponent` necessary?

File: website/static/js/fangorn.js [hunk 4]
```diff
@@ -209,30 +213,30 @@ function resolveIconView(item) {
...
         if (item.data.isRegistration) {
             return returnView('registeredComponent', item.data.category);
         }
         return returnView('component', item.data.category);
     }
 
-    if (item.data.nodeType === 'pointer') {
+    if (item.data.nodeType === 'pointer' || item.data.isPointer) {
```
Comment (sloria) [inline website/static/js/fangorn.js]: Related to previous comment--`nodeType` can never be `pointer`; it can only ever be `project` or `component`.

**Comment by sloria** [comment]
Pass finished. :point_up_2:

**Comment by leb2dg** [comment]
Updated! You were correct @sloria, those were redundant checks. :grimacing:

File: website/static/js/projectorganizer.js [hunk 5]
```diff
@@ -324,9 +324,6 @@ function _poToggleCheck(item) {
  */
 function _poResolveIcon(item) {
     var newIcon = Fangorn.Utils.resolveIconView(item);
-    if (newIcon === null) {
```
Comment (sloria) [inline website/static/js/projectorganizer.js]: With the removal of this code, `_poResolveIcon` is exactly equivalent to `Fangorn.Utils.resolveIconView`. Therefore, can we just set `resolveIcon` in Treebeard to `Fangorn.Utils.resolveIconView`?

**Comment by sloria** [comment]
One more suggested change and this is good to merge. :v:

**Comment by leb2dg** [comment]
Good call! I removed `_poResolveIcon`.

**Comment by sloria** [comment]
Looks good. :octopus:

**Comment by nicipfeiffer** [comment]
@sloria Is this ready for QA? As well as #2871?

**Comment by sloria** [comment]
@nicipfeiffer Yes, those PRs are ready for QA

**Pull request merged**